src/views/search/orama.tsx

```diff
@@ -0,0 +1,189 @@
+import type { TypedDocument, Orama, Results, Nullable } from "@orama/orama";
+import { OramaClient, SortByClauseUnion } from "@oramacloud/client";
+import { PostInfo } from "types/PostInfo";
+import { CollectionInfo } from "types/CollectionInfo";
+import { PropsWithChildren } from "components/types";
+import { createContext } from "preact";
+import { useContext } from "preact/hooks";
+import { SearchQuery } from "./search";
+
+const postSchema = {
+	slug: "string",
+	tags: "enum[]",
+	authors: "enum[]",
+	title: "string",
+	excerpt: "string",
+	description: "string",
+	searchMeta: "string",
+	publishedTimestamp: "number",
+} as const;
+
+const collectionSchema = {
+	slug: "string",
+	title: "string",
+	excerpt: "string",
+	description: "string",
+	searchMeta: "string",
+	tags: "enum[]",
+	authors: "enum[]",
+	publishedTimestamp: "number",
+} as const;
+
+type PostDocument = TypedDocument<Orama<typeof postSchema>>;
+type CollectionDocument = TypedDocument<Orama<typeof collectionSchema>>;
+
+export interface SearchContext {
+	postClient: OramaClient;
+	collectionClient: OramaClient;
+}
+
+export const SearchClient = createContext<SearchContext>(undefined as never);
+
+interface OramaClientProviderProps extends PropsWithChildren {
+	params?: Partial<ConstructorParameters<typeof OramaClient>[0]>;
+}
+
+export function OramaClientProvider(props: OramaClientProviderProps) {
+	const postClient = new OramaClient({
+		endpoint: "https://cloud.orama.run/v1/indexes/playful-programming-p9lpvl",
+		api_key: "OLeHrFPWLR0alSSZkMiq4tokMZZNEbDL",
+		...props.params
+	});
+
+	const collectionClient = new OramaClient({
+		endpoint:
+			"https://cloud.orama.run/v1/indexes/playful-programming-collections-oksaw0",
+		api_key: "yxsPLU2kjhAjjNxTUQ4F6c3bF9eXUSNJ",
+		...props.params
+	});
+
+	const context: SearchContext = { postClient, collectionClient };
+
+	return (
+		<SearchClient.Provider value={context}>
+			{props.children}
+		</SearchClient.Provider>
+	);
+}
+
+export function useOramaSearch() {
+	const searchCtx = useContext(SearchClient);
+	return {
+		searchForTerm: (query: SearchQuery, signal: AbortSignal) => searchForTerm(searchCtx, query, signal),
+	};
+}
+
+/**
+ * Given two Record<string, number>, combines them into one record, summing any overlapping entries.
+ * Ex. addMerge({a:1,b:2}, {b:1}) -> {a:1,b:3}
+ */
+function addMerge(obj1: Record<string, number>, obj2: Record<string, number>): Record<string, number> {
+	return Object.entries(obj2).reduce(
+		(acc, [key, value]) => ({ ...acc, [key]: (acc[key] || 0) + value }),
+		{ ...obj1 },
+	);
+}
+
+export async function searchForTerm({ postClient, collectionClient }: SearchContext, query: SearchQuery, signal: AbortSignal) {
+	// Schema should be passed to `search` method when:
+	// https://github.com/askorama/oramacloud-client-javascript/pull/35
```

Comment:
This was merged. We can change this now

Reply:
This doesn't appear to be working. I'm only seeing this function in the type definitions, and not the version with generics (using the latest version):
```ts
search(query: ClientSearchParams, config?: SearchConfig): Promise<Nullable<Results<AnyDocument>>>;
```

I'll update this comment & the client lib version, but will look into what's happening with the types later. (I don't think it should block this PR)